manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -9,26 +9,29 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CatalogRequirements contain the specifics for catalogging the data asset
-type CatalogRequirements struct {
-	// CatalogService specifies the datacatalog service that will be used for catalogging the data into.
-	// +optional
-	CatalogService string `json:"service,omitempty"`
-
+// TargetCatalog contain the specifics for the catalog which needs to be updated.
+// Updates may be adding a new dataset to the catalog (currently supported) or deleting an existing dataset (future)
+type TargetCatalog struct {
```

Comment:
When deleting an existing dataset, the dataset id contains all relevant information including catalog id.
Regarding adding a new asset, are we assuming the catalog already exists? Do all catalogs support a catalog concept? For example, katalog - is this going to be a namespace? 
We might need more data for the registration. I suggest to keep it in the designated structure instead of globally.

Reply:
We will have several modes for write:
(1) Write a new dataset - fybrik allocates storage and registers a new entry in the catalog.  Catalog and filename in datasetid, and flowparams.isnewdataset = true
(2) Write a new dataset where storage has already been allocated by the user.  User creates the new catalog entry and indicates it to Fybrik in DataSetID.  No need for Fybrik to not register anything in the catalog.
(3) Write to existing dataset - Same as #2 except that the user doesn't have to create a new catalog entry.

FlowParams is intentionally separate so we can expand it.  I'm hoping to avoid having structure per dataflow type

Reply:
(2) and (3) are similar from the spec perspective. DatasetID points to the asset metadata + connection, so all is set.
I don't understand (1). Fybrik will allocate storage. For registration, it will need a full/partial identifier of the catalog/asset. It will also need to register the metadata (credentials come from Fybrik). How? Do you mean that in this case users will register the new metadata in some asset yaml and point to this yaml in DatasetID while specifying that this is not the real dataset id? 
Copy (explicit) is even more challenging. DatasetID will point to a real asset that needs to be copied using allocated storage or not, and registered or not. How will you specify all this information?

Reply:
@roytman I have a separate issue for addressing adding metadata / tags.  It will have to be provided by the user ... unless we had a module that runs and analyzes the data.  That's out of scope though

Reply:
> (2) and (3) are similar from the spec perspective. DatasetID points to the asset metadata + connection, so all is set. I don't understand (1). Fybrik will allocate storage. For registration, it will need a full/partial identifier of the catalog/asset. It will also need to register the metadata (credentials come from Fybrik). How? Do you mean that in this case users will register the new metadata in some asset yaml and point to this yaml in DatasetID while specifying that this is not the real dataset id? 

No.  The user will provide in DatasetID something like ex: HRCatalog/JanSalaries.  This is not a unique identifier.  It tells Fybrik that HRCatalog is the catalog in which the file JanSalaries should be registered.  Fybrik knows this is not a unique ID because flowparams.isnewdataset = true


Copy (explicit) is even more challenging. DatasetID will point to a real asset that needs to be copied using allocated storage or not, and registered or not. How will you specify all this information?

DatasetID will be the catalog id of source data set (the same way we did it for ingest).  If the copy needs to be registered in the catalog then DataContext.FlowParams.Catalog will contain the name of the catalog in which to register the copy.  Name of the dataset will be copied from the source.  If DataContext.FlowParams.Catalog is not indicated, then the asset won't be cataloged (not a good idea though).